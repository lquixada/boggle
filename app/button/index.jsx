/*
 * Control
 */
var React = require('react');
var style = require('./index.less');


var Button = React.createClass({
  render: function () {
    return (
      <div id="control">
        <button type="button" onClick={this.props.onClick}>{this.props.started?'stop!':'start!'}</button>
      </div>
    );
  }
});

module.exports = Button;
