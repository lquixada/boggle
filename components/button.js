import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';


class Button extends React.Component {
  render() {
    return (
      <div id="button">
        {this.renderButton()}
      </div>
    );
  }

  renderButton() {
    if (this.props.started) {
      return <button type="button" onClick={this.props.stopGame}>stop</button>;
    } else {
      return <button type="button" onClick={this.props.startGame}>start</button>;
    }
  }
}

const mapStateToProps = (state) => ({
  started: state.started
});

export default connect(mapStateToProps, actions)(Button);
