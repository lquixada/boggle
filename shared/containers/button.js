import React from 'react';
import Button from '../components/button';
import {connect} from 'react-redux';
import * as actions from '../actions';

class ButtonContainer extends React.Component {
  getText() {
    if (this.props.started) {
      return 'stop';
    }
    
    return 'start';
  }

  handleClick() {
    if (this.props.started) {
      this.props.stopGame();
    } else {
      this.props.startGame();
    }
  }

  render() {
    return (
      <Button onClick={this.handleClick.bind(this)} text={this.getText()} />
    );
  }
}

const mapStateToProps = ({started}) => ({started});
export default connect(mapStateToProps, actions)(ButtonContainer);
