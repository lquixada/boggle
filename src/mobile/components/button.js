import React from 'react';
import {StyleSheet, Text, TouchableHighlight} from 'react-native';

class Button extends React.Component {
  render() {
    return (
      <TouchableHighlight
        onPress={this.props.onClick}
        style={styles.wrapper}
        underlayColor="#286f23"
      >
        <Text style={styles.text}>{this.props.text}</Text>
      </TouchableHighlight>
    );
  }
}

const styles = StyleSheet.create({
  wrapper: {
    width: 100,
    height: 44,
    padding: 10,
    alignItems: 'center',
    borderRadius: 5,
    backgroundColor: '#6c6',
  },
  text: {
    color: '#fff',
    fontSize: 20,
  },
});

export default Button;
