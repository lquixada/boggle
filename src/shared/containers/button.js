import React from 'react';
import {connect} from 'react-redux';

import Button from '../components/button';
import * as actions from '../actions';

export class ButtonContainer extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  getText() {
    if (this.props.started) {
      return 'stop';
    }

    return 'start';
  }

  handleClick() {
    if (this.props.started) {
      this.props.stopGame();
    } else {
      this.props.startGame();
    }
  }

  render() {
    return (
      <Button onClick={this.handleClick} text={this.getText()} />
    );
  }
}

const mapStateToProps = ({started}) => ({started});
export default connect(mapStateToProps, actions)(ButtonContainer);
