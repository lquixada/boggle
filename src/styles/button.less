#button {
  button {
    padding: 1rem;
    min-width: 10rem;
    border: none;
    cursor: pointer;
    color: #fff;
    font-size: 2rem;
    line-height: normal;
    text-align: center;
    vertical-align: middle;
    outline: none;
    border-radius: .4rem;
    background-color: #6c6;
    box-sizing: border-box;

    &:hover {
      background-color: #6c6;
      text-decoration: none !important;
    }

    &:active {
      background-color: #286f23;
      -webkit-box-shadow: inset 0 0.2rem 0.2rem rgba(0, 0, 0, 0.2);
      -moz-box-shadow: inset 0 0.2rem 0.2rem rgba(0, 0, 0, 0.2);
      box-shadow: inset 0 0.2rem 0.2rem rgba(0, 0, 0, 0.2);
    }
  }
}
