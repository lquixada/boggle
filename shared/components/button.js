import '../styles/button.less';
import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';


class Button extends React.Component {
  render() {
    return (
      <div id="button">
        {this.renderButton()}
      </div>
    );
  }

  renderButton() {
    if (this.props.started) {
      return <button type="button" onClick={this.props.stopGame}>stop</button>;
    } else {
      return <button type="button" onClick={this.props.startGame}>start</button>;
    }
  }
}

const mapStateToProps = ({ started }) => ({ started });

export default connect(mapStateToProps, actions)(Button);
