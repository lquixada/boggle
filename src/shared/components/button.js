import React from 'react';
import PropTypes from 'prop-types';

import '../../styles/button.scss';

const Button = ({onClick, text}) => (
  <div id="button">
    <button type="button" onClick={onClick}>{text}</button>
  </div>
);

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired
};

export default Button;
