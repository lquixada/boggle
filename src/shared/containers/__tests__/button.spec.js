import React from 'react'
import { shallow } from 'enzyme'

import { ButtonContainer } from '../button'

describe('<ButtonContainer />', () => {
  it('shows "start" by default', () => {
    const component = shallow(<ButtonContainer />)
    expect(component).toMatchSnapshot()
  })

  it('shows "stop" when the game has started', () => {
    const component = shallow(<ButtonContainer started />)
    expect(component).toMatchSnapshot()
  })

  it('shows "start" when the game has stopped', () => {
    const component = shallow(<ButtonContainer started={false} />)
    expect(component).toMatchSnapshot()
  })

  it('toggles state from "start" to "stop" when clicked', () => {
    const startGame = jest.fn()
    const component = shallow(<ButtonContainer started={false} startGame={startGame} />)
    component.simulate('click')

    expect(startGame).toHaveBeenCalled()
  })

  it('toggles state from "stop" to "start" when clicked', () => {
    const stopGame = jest.fn()
    const component = shallow(<ButtonContainer started stopGame={stopGame} />)
    component.simulate('click')

    expect(stopGame).toHaveBeenCalled()
  })
})
