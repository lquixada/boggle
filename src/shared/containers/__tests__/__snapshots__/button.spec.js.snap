// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonContainer /> shows "start" by default 1`] = `
<ButtonArea
  onClick={[Function]}
  text="start"
/>
`;

exports[`<ButtonContainer /> shows "start" when the game has stopped 1`] = `
<ButtonArea
  onClick={[Function]}
  text="start"
/>
`;

exports[`<ButtonContainer /> shows "stop" when the game has started 1`] = `
<ButtonArea
  onClick={[Function]}
  text="stop"
/>
`;
